File: ch02-03/ord_list.py
def linear_search(array: list, search_value: int) -> int:
    for index, val in enumerate(array):
        if val == search_value:
            return index
        elif val > search_value:
            break

def binary_search(array: list, search_value: int) -> int:
    lower_bound = 0
    upper_bound = len(array) - 1
    while (lower_bound <= upper_bound):
        midpoint = int((upper_bound + lower_bound) / 2)

        value_at_midpoint = array[midpoint]
        if search_value == value_at_midpoint:
            return midpoint
        elif search_value < value_at_midpoint:
            upper_bound = midpoint - 1
        else:
            lower_bound = midpoint + 1

File: ch02-03/ord_list.rb
def linear_search(array, search_value)
    array.each_with_index do |element, index|
        if element == search_value
            return index
        elsif element > search_value
            break
        end
    end

    return nil
end

print(chars.ascii_only?(),"\n")
p linear_search([3, 17, 75, 80, 202], 202)
